File: sort-array-by-parity/sort-array-by-parity.go
func sortArrayByParity(nums []int) []int {
    arr := make([]int, len(nums))
	copy(arr, nums)

	evenPtr := 0
	oddPtr := len(arr) - 1
	for _, num := range nums {
		if num%2 == 0 {
			arr[evenPtr] = num
			evenPtr++
		} else {
			arr[oddPtr] = num
			oddPtr--
		}
	}

	return arr
}

File: sort-array-by-parity/sort-array-by-parity.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int[] arr = new int[nums.length];
        int evenPtr = 0;
        int oddPtr = arr.length-1;
        for (int num : nums) {
            if (num % 2 == 0) {
                arr[evenPtr++] = num;
            } else {
                arr[oddPtr--] = num;
            }
        }

        return arr;
    }
}